### PR TITLE
refactor: Enhance the basic validation for MsgDeployFungibleCoinZRC20

### PR DESCRIPTION
**Pull request by lukema95**
# Description

Add a check for the `Decimals` field in the basic validation of MsgDeployFungibleCoinZRC20.

Closes: #1295

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [x] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [x] I have added unit tests that prove my fix feature works

### CODE REVIEW

**Review by lumtis — APPROVED** [review]
LGTM

**Review by kingpinXD — APPROVED** [review]

File: x/fungible/types/message_deploy_fungible_coin_zrc20.go [hunk 1]
```diff
@@ -50,7 +50,11 @@ func (msg *MsgDeployFungibleCoinZRC20) ValidateBasic() error {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
 	}
 	if msg.GasLimit < 0 {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidGasLimit, "invalid gas limit (%s)", err)
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidGasLimit, "invalid gas limit")
+	}
+
+	if msg.Decimals > 255 {
```
Comment (ws4charlie) [inline x/fungible/types/message_deploy_fungible_coin_zrc20.go:56]: Would `if msg.Decimals > 77` be a more realistic number in here as ERC20 standard uses `uint256` `(2 ** 256)-1 ≈ 10 ** 77` ?

Comment (ws4charlie) [reply x/fungible/types/message_deploy_fungible_coin_zrc20.go:56]: @lukema95 what do you think

Comment (lukema95) [reply x/fungible/types/message_deploy_fungible_coin_zrc20.go:56]: I agree with you, `if msg.Decimals > 77` would be more accurate, I will change it later.

**Pull request merged**